library/core/src/ptr/non_null.rs

```diff
@@ -430,6 +432,10 @@ impl<T: ?Sized> NonNull<T> {
     #[must_use = "this returns the result of the operation, \
                   without modifying the original"]
     #[inline]
+    // Address preservation
+    #[ensures(|result: &NonNull<U>| result.as_ptr() as *const () as usize == self.as_ptr() as *const () as usize )]
```

Comment:
Please use `addr()` instead wherever you are casting to `usize`.

Reply:
I have changed it. Thanks!